test/redis/interval_cache_test.exs

```diff
@@ -17,36 +18,38 @@ defmodule Castle.RedisIntervalCacheTest do
   end
 
   test "assembles keys for timestamps", %{from: from, to: to} do
-    keys = interval_keys(@prefix, from, to, 900)
+    keys = interval_keys(@prefix, from, to, QuarterHourly)
     assert length(keys) == 6
-    assert hd(keys) == "#{@prefix}.900.2017-03-22T01:15:00Z"
-    assert List.last(keys) == "#{@prefix}.900.2017-03-22T02:30:00Z"
+    assert hd(keys) == "#{@prefix}.15MIN.2017-03-22T01:15:00Z"
+    assert List.last(keys) == "#{@prefix}.15MIN.2017-03-22T02:30:00Z"
   end
 
   test "sets a separate ttl for current intervals" do
     now = Timex.now
-    past = Timex.shift(Timex.now, seconds: -7200)
-    later = Timex.shift(Timex.now, seconds: 100)
-
-    assert interval_ttls(past, now, 900) == [43200, 43200, 43200, 43200, 15, 15, 15, 15]
-    assert interval_ttls(past, now, 3600) == [43200, 15]
-    assert interval_ttls(past, now, 86400) == [15]
-
-    assert interval_ttls(past, later, 900) == [43200, 43200, 43200, 43200, 15, 15, 15, 15, 15]
-    assert interval_ttls(past, later, 3600) == [43200, 15, 15]
-    assert interval_ttls(past, later, 86400) == [15]
+    past = Timex.shift(now, seconds: -7200)
+    later = Timex.shift(now, seconds: 901)
+
+    assert interval_ttls(past, now, QuarterHourly) == [2592000, 2592000, 2592000, 2592000, 300, 300, 300, 300, 300]
+    assert interval_ttls(past, now, Hourly) == [2592000, 300, 300]
+    assert interval_ttls(past, now, Daily) == [300]
+    assert interval_ttls(past, now, Weekly) == [300]
+    assert interval_ttls(past, now, Monthly) == [300]
+
+    assert interval_ttls(past, later, QuarterHourly) == [2592000, 2592000, 2592000, 2592000, 300, 300, 300, 300, 300, 300]
+    assert interval_ttls(past, later, Hourly) == [2592000, 300, 300]
```

Comment:
oops, approved too soon -- this is failing for me
```elixir
 test sets a separate ttl for current intervals (Castle.RedisIntervalCacheTest)
     test/redis/interval_cache_test.exs:27
     Assertion with == failed
     code:  interval_ttls(past, later, Hourly) == [2592000, 300, 300]
     left:  [2592000, 300, 300, 300]
     right: [2592000, 300, 300]
     stacktrace:
       test/redis/interval_cache_test.exs:39: (test)
```

Reply:
Yeah, that one's been giving me trouble.  It only passes at certain times of day!  Will fix.

(Mocking the current time in Elixir wasn't immediately easy ... so of course i gave up!)